processor/aggregationprocessor/config.go

```diff
@@ -48,16 +53,45 @@ func (a AggregateConfig) Validate() error {
 		return fmt.Errorf("invalid aggregate type for `type`: %s", a.Type)
 	}
 
+	_, err := a.MetricNameExpression()
+	if err != nil {
+		return fmt.Errorf("failed to parse metric_name_expression: %w", err)
+	}
+
 	return nil
 }
 
-// MetricNameString returns the configured name for the emitted metric, or "$0" if none was specified.
-func (a AggregateConfig) MetricNameString() string {
-	if a.MetricName != "" {
-		return a.MetricName
+const metricNameKey = "metric_name"
```

Comment:
nit: any reason this is middle of the file rather than the top? Just thinking consistency wise with most of our code.

Reply:
I put it there since it's close to actual usage, but good point. Moved to top.